client/data/media/use-update-media-mutation.js

```diff
@@ -0,0 +1,20 @@
+import { useCallback } from 'react';
+import { useMutation } from 'react-query';
+import wp from 'calypso/lib/wp';
+
+export const useUpdateMediaMutation = ( queryOptions = {} ) => {
+	const mutation = useMutation(
+		( { siteId, mediaId, updates } ) =>
+			wp.req.post( `/sites/${ siteId }/media/${ mediaId }`, updates ),
+		{ ...queryOptions }
```

Comment:
Can just pass `queryOptions` without shallow cloning or defaulting.

Reply:
fixed in https://github.com/Automattic/wp-calypso/pull/58604/commits/a5b6e8a3348499acc17c9fc641b9ef7284c93230